org.metafacture.fix/src/main/java/org/metafacture/metafix/FixMethod.java

```diff
@@ -0,0 +1,448 @@
+            default:
+                nestedMap = new LinkedHashMap<>();
+                nestedList.add(nestedMap);
+                insert(mode, nestedMap, remainingKeys, value);
+                break;
+        }
+    }
+
+    @SuppressWarnings("checkstyle:ReturnCount")
+    static Object find(final Map<String, Object> map, final String[] keys) {
+        final String currentKey = keys[0];
+        if (!map.containsKey(currentKey)) {
+            return null;
+        }
+        if (keys.length == 1) {
+            return map.get(currentKey);
+        }
+        final String[] remainingKeys = Arrays.copyOfRange(keys, 1, keys.length);
+        return findNested(map, currentKey, remainingKeys);
+    }
+
+    private static Object findNested(final Map<String, Object> map, final String currentKey,
+            final String[] remainingKeys) {
+        final Object nested = map.get(currentKey);
+        // TODO: array of maps, like in insertNested
+        if (nested instanceof List) {
+            return ((List<?>) nested).stream().map(o -> findNested(map, currentKey, remainingKeys))
+                    .collect(Collectors.toList());
+        }
+        if (nested instanceof Map) {
+            @SuppressWarnings("unchecked")
+            final Object result = find((Map<String, Object>) nested, remainingKeys);
+            return result;
+        }
+        throw new IllegalStateException(NESTED + nested);
+    }
+
+    private static Object remove(final Map<String, Object> map, final String[] keys) {
+        final String currentKey = keys[0];
+        if (keys.length == 1) {
+            map.remove(currentKey);
+        }
+        if (!map.containsKey(currentKey)) {
+            return map;
+        }
+        final String[] remainingKeys = Arrays.copyOfRange(keys, 1, keys.length);
+        return removeNested(map, currentKey, remainingKeys);
+    }
+
+    private static Object removeNested(final Map<String, Object> map, final String currentKey,
+            final String[] remainingKeys) {
+        final Object nested = map.get(currentKey);
+        if (!(nested instanceof Map)) {
+            throw new IllegalStateException(NESTED + nested);
+        }
+        @SuppressWarnings("unchecked")
+        final Object result = remove((Map<String, Object>) nested, remainingKeys);
+        return result;
+    }
+
+    private static void copy(final Map<String, Object> record, final List<String> params) {
+        final String oldName = params.get(0);
+        final String newName = params.get(1);
+        final Object value = find(record, split(oldName));
+        if (value != null) {
+            final List<Object> vs = Metafix.asList(value);
+            for (final Object v : vs.stream().filter(v -> v != null).collect(Collectors.toList())) {
+                insert(InsertMode.APPEND, record, split(newName), v.toString());
+            }
+        }
+    }
+
+    static String[] split(final String s) {
+        return s.split("\\.");
```

Comment:
Is this for consistency with Metamorph only or do you have a specific use case?

Since Catmandu uses dot notation, and dot notation is very common, I'd prefer to keep it simple here.

Reply:
Yes, it's mostly about consistency for now. I haven't had a specific need for this so far, but as soon as you have data with periods in their fields you're stuck (SISIS comes to mind, although we probably won't have to deal with those anymore).

I'm totally fine with postponing this; that's why I created the ticket. We can discuss there if it's worth the effort.

(More or less everything I noted/suggested/questioned here is kind of optional; what doesn't get addressed immediately could/should be turned into tickets for further discussion and prioritization.)

Reply:
> what doesn't get addressed immediately could/should be turned into tickets for further discussion and prioritization

:+1: